### DISCUSSION TITLE
WiFiManager not updates DNS

### DISCUSSION THREAD

**Discussion by Gandalph65** [khoih-prog/ESP_WiFiManager#104] [Q&A]
If change DNS on the Config Portal the DNS1 and DNS2 never gonna updated.
Still use after reboot the ESP32 device compiled DNS1 and DNS2 IP from ESPAsync_WiFiManager.hpp

**Comment by khoih-prog** [comment]
Hi @Gandalph65 

Please check there is something wrong in your code.

I'm testing with [**ConfigOnDoubleReset**](https://github.com/khoih-prog/ESP_WiFiManager/tree/master/examples/ConfigOnDoubleReset) example and DNS is modified as expected

### Entering CP with stored old DNS2 = `8.8.8.8`

```
Starting ConfigOnDoubleReset with DoubleResetDetect using LittleFS on ESP32_DEV
ESP_WiFiManager v1.12.1
ESP_DoubleResetDetector v1.3.2
ESP Self-Stored: SSID = HueNet, Pass = 12345678
[WM] * Add SSID =  HueNet , PW =  12345678
Got ESP Self-Stored Credentials. Timeout 120s for Config Portal
[WM] LoadWiFiCfgFile 
[WM] OK
[WM] stationIP = 0.0.0.0 , gatewayIP = 192.168.2.1
[WM] netMask = 255.255.255.0
[WM] dns1IP = 192.168.2.1 , dns2IP = 8.8.8.8
Got stored Credentials. Timeout 120s for Config Portal
[WM] Current TZ_Name = America/New_York , TZ =  EST5EDT,M3.2.0,M11.1.0
LittleFS Flag read = 0xD0D01234
doubleResetDetected
Saving config file...
Saving config file OK
Open Config Portal without Timeout: Double Reset Detected
Starting configuration portal @ 192.168.4.1:80, SSID = ESP_91248, PWD = MyESP_91248
```

---

### After CP with new DNS2 = `4.4.4.4`

```
Starting ConfigOnDoubleReset with DoubleResetDetect using LittleFS on ESP32_DEV
ESP_WiFiManager v1.12.1
ESP_DoubleResetDetector v1.3.2
ESP Self-Stored: SSID = HueNet, Pass = 12345678
[WM] * Add SSID =  HueNet , PW =  12345678
Got ESP Self-Stored Credentials. Timeout 120s for Config Portal
[WM] LoadWiFiCfgFile 
[WM] OK
[WM] stationIP = 0.0.0.0 , gatewayIP = 192.168.2.1
[WM] netMask = 255.255.255.0
[WM] dns1IP = 192.168.2.1 , dns2IP = 4.4.4.4
Got stored Credentials. Timeout 120s for Config Portal
[WM] Current TZ_Name = America/New_York , TZ =  EST5EDT,M3.2.0,M11.1.0
LittleFS Flag read = 0xD0D04321
No doubleResetDetected
Saving config file...
Saving config file OK
[WM] * Add SSID =  HueNet , PW =  12345678
[WM] * Add SSID =  HueNet2 , PW =  12345678
ConnectMultiWiFi in setup
[WM] ConnectMultiWiFi with :
[WM] * Flash-stored Router_SSID =  HueNet , Router_Pass =  12345678
[WM] * Add SSID =  HueNet , PW =  12345678
[WM] * Additional SSID =  HueNet , PW =  12345678
[WM] * Additional SSID =  HueNet2 , PW =  12345678
[WM] Connecting MultiWifi...
[WM] WiFi connected after time:  1
[WM] SSID: HueNet ,RSSI= -24
[WM] Channel: 10 ,IP address: 192.168.2.130
After waiting 11.14 secs more in setup(), connection result is connected. Local IP: 192.168.2.130
Stop doubleResetDetecting
Saving config file...
Saving config file OK
```

**Comment by Gandalph65** [comment]
Hi Khoi Hoang,

thanks for your reply.

Yes i got some Code Issues.

I been trying your Sample Async_ConfigPortalParamsOnSwitch

but it toke me several day to got PlatformIO run with it.

Reason your PlatformIO.ini sample isnt working.

so I've been trvying bigger number of diffrent Libraries. Fork some try to change them

And started to change the code (even more than required for my own Homenetwork)

Finaly i got the right Version of Espressif run (for both ESP8266 and ESP32) and all needed Libs, so compile completed. Got access to the Configportal with even those fault I've been asking you.
You some of those Library Issues i got help from PlatformIO Community.

At least i started over with a new copy this sample and it worked fine.

If you like i could provide you mine PlatformIO.ini with all information get the ESP32 and ESP8266 run.

At the Moment i working on merge this Sample with your ESP_AsyncFSBrowser.

For my tests i using ESP8266 12e and an ESP32 DEVKIT V1

 

best regards

 

Joachim Scheffel (Gandalph65)
Germany

 
 

Gesendet: Freitag, 06. Januar 2023 um 01:47 Uhr
Von: "Khoi Hoang" ***@***.***>
An: "khoih-prog/ESP_WiFiManager" ***@***.***>
Cc: "Gandalph65" ***@***.***>, "Mention" ***@***.***>
Betreff: Re: [khoih-prog/ESP_WiFiManager] WiFiManager not updates DNS (Discussion #104)


 

Hi @Gandalph65

Please check there is something wrong in your code.

I'm testing with ConfigOnDoubleReset example and DNS is modified as expected

Entering CP with stored old DNS2 = 8.8.8.8

Starting ConfigOnDoubleReset with DoubleResetDetect using LittleFS on ESP32_DEV
ESP_WiFiManager v1.12.1
ESP_DoubleResetDetector v1.3.2
ESP Self-Stored: SSID = HueNet, Pass = 12345678
[WM] * Add SSID =  HueNet , PW =  12345678
Got ESP Self-Stored Credentials. Timeout 120s for Config Portal
[WM] LoadWiFiCfgFile
[WM] OK
[WM] stationIP = 0.0.0.0 , gatewayIP = 192.168.2.1
[WM] netMask = 255.255.255.0
[WM] dns1IP = 192.168.2.1 , dns2IP = 8.8.8.8
Got stored Credentials. Timeout 120s for Config Portal
[WM] Current TZ_Name = America/New_York , TZ =  EST5EDT,M3.2.0,M11.1.0
LittleFS Flag read = 0xD0D01234
doubleResetDetected
Saving config file...
Saving config file OK
Open Config Portal without Timeout: Double Reset Detected
Starting configuration portal @ 192.168.4.1:80, SSID = ESP_91248, PWD = MyESP_91248



After CP with new DNS2 = 4.4.4.4

Starting ConfigOnDoubleReset with DoubleResetDetect using LittleFS on ESP32_DEV
ESP_WiFiManager v1.12.1
ESP_DoubleResetDetector v1.3.2
ESP Self-Stored: SSID = HueNet, Pass = 12345678
[WM] * Add SSID =  HueNet , PW =  12345678
Got ESP Self-Stored Credentials. Timeout 120s for Config Portal
[WM] LoadWiFiCfgFile
[WM] OK
[WM] stationIP = 0.0.0.0 , gatewayIP = 192.168.2.1
[WM] netMask = 255.255.255.0
[WM] dns1IP = 192.168.2.1 , dns2IP = 4.4.4.4
Got stored Credentials. Timeout 120s for Config Portal
[WM] Current TZ_Name = America/New_York , TZ =  EST5EDT,M3.2.0,M11.1.0
LittleFS Flag read = 0xD0D04321
No doubleResetDetected
Saving config file...
Saving config file OK
[WM] * Add SSID =  HueNet , PW =  12345678
[WM] * Add SSID =  HueNet2 , PW =  12345678
ConnectMultiWiFi in setup
[WM] ConnectMultiWiFi with :
[WM] * Flash-stored Router_SSID =  HueNet , Router_Pass =  12345678
[WM] * Add SSID =  HueNet , PW =  12345678
[WM] * Additional SSID =  HueNet , PW =  12345678
[WM] * Additional SSID =  HueNet2 , PW =  12345678
[WM] Connecting MultiWifi...
[WM] WiFi connected after time:  1
[WM] SSID: HueNet ,RSSI= -24
[WM] Channel: 10 ,IP address: 192.168.2.130
After waiting 11.14 secs more in setup(), connection result is connected. Local IP: 192.168.2.130
Stop doubleResetDetecting
Saving config file...
Saving config file OK
